project-reports/2023/2023-Q4-Hyperledger-Indy.md

```diff
@@ -0,0 +1,173 @@
+in the last quarterly report. The event was again well attended, 40+
+participants, and the topics covered included specific actions for moving the
+project forward. Here is a
+[link](https://docs.google.com/document/d/1Gkd-V17tAYQFI4ymtIGUjfQxBlgOYVj0WBdX1ZdWA_g/edit?usp=sharing)
+to the materials from both summits. See the [Overall
+Activity](#overall-activity-in-the-past-quarter) for more details of the
+discussions.
+
+Per the [Indy Quarterly Activity Dashboard], there were 149 commits
+(up about 50% from last quarter) from 11 contributors (down about 50% from last quarter).
+
+[Indy Quarterly Activity Dashboard]: https://insights.lfx.linuxfoundation.org/projects/hyperledger%2Findy/dashboard;subTab=technical?time=%7B%22from%22:%222023-07-01T08:00:00.000Z%22,%22type%22:%22absolute%22,%22to%22:%222023-09-30T07:00:00.000Z%22%7D
+
+## Questions/Issues for the TOC
+
+See the idea in that came out of the Second Hyperledger Indy Ecosystem Summit:
+
+> Is there a way for the Hyperledger Foundation to enable a group to collect
+funds for an open source objective, and the use those funds to enable a code
+contribution?
+
+### Issues from previous reports
+
+#### **Diversity of Contributor Community**
+
+See the updated type of information in the appropriate section of this report.
+
+## Releases
+
+- indy-vdr -  - v0.4.0-dev.13 to v0.4.0-dev.16
+- indy-node-container - v1.2.5
+- indy-shared-rs - v0.3.3
+
+## Overall Activity in the Past Quarter
+
+GitHub activity was focused on the so-called Shared Components ([indy-vdr],
+[indy-shared-rs] and [aries-askar]) and the transition away from the Indy SDK. The
+maintainer community has discussed the formal deprecation of the Indy SDK and
+the Aries maintainers have been busy making the Shared Components the default in
+Aries, in preparation for the deprecation announcement.  
+
+At the Second Indy Ecosystem Summit, two discussions focused on paths forward.
+On the technical side, [DSR Corporation](https://en.dsr-corporation.com/)
+presented several ideas for evolving the Indy Node ledger component, including
+evolving the current implementation, or implementing Indy transactions on
+another network, such as [Hyperledger Besu](https://www.hyperledger.org/projects/besu), or
+[Cosmos](https://cosmos.network/). While there is no pressing feature set needed in the
+foreseeable future, the lack of maintainers on the existing code base is concerning. Those
+with an opinion thought that Besu was the most interesting alternative and work has
+started on defining what that might mean. This breaks into a couple of questions:
+
+* How to write Indy (or more accurately, AnonCreds) objects on a Besu network
+  (schema, credential definitions, and revocation registries)?
+* Can and should the existing Indy roles be used when running on a Besu network?
+* Can and should the existing Indy authorization rules be used when running on a Besu network?
```

Comment:
Why the "should"? Are there drawbacks in the Indy permission model?

If you can port Indy objects to Besu, I imagine it should not be too hard to port it to any other kind of ledger, right?

Reply:
The “should” is just there in case it is difficult to achieve and there are not sufficient benefits. The community has adapted to the Indy roles/permission model, and definitely works. However, I didn’t want to be so rigid as to say “It must be exactly the same or we walk away”.  I wanted to leave room for other ideas, and potential learnings from Besu and its deployments.  That said — in subsequent discussions with some Besu operators, I see that the permissions/roles are quite similar.

The objects on Indy are the objects of Hyperledger AnonCreds, and that project has already demonstrated that those objects can be stored on other types of ledgers. The point of moving AnonCreds out of Indy was to promote that the technology is ledger-agnostic, and to make it easy to root AnonCreds credentials on any ledger (and even non-ledger) storage.

The goal of an “Indy on Besu” approach is to provide a transition from current Indy ledgers to future Indy ledgers — providing comparable tooling and capabilities that exist for Indy to others wanting to run a ledger that has first class support for privacy-preserving verifiable credentials. The benefit of that goal is that ANY Besu network would gain that “first class support”.